### ISSUE TITLE
[Bug] Sidebar pages in Vivaldi browser get broken dark theme

### ISSUE THREAD

**Issue by jenda69** [darkreader/darkreader#6842]
# Bug Report

## Bug Description

All pages in the sidebar in the Vivaldi browser get some kind of partial dark theme. (Probably since last update?)

## Website Address

Any, but it must be in the sidebar in the Vivaldi

## Steps To Reproduce

- Launch Vivaldi
- Install Dark Reader
- (optionally) set Dark Reader to "Disabled on all websites by default"
- Add any page to sidebar panel (e.g. Instagram, Duolingo...) 

## Expected Behavior

I'd expect Dark Reader to not work in the sidebar at all since extensions (like uBlock etc.) usually don't work there. And it didn't work there until today. But if it does work, it should display websites correctly and also adhere to Dark Reader settings.

## Actual Behavior

Websites get dark background, white text, but lot of images are missing (on Instagram and Duolingo). I also changed DR settings to "Disabled on all websites by default" and even non-whitelisted pages in the sidebar have this broken dark theme anyway.

## Screenshots

Duolingo missing all the icons for the lessons.
![Duolingo](https://i.imgur.com/M7CXS7m.png)

## System Information

- OS: Windows 10 Pro 20H2 64-bit
- Browser: latest Vivaldi 4.2.2406.48 64-bit
- Dark Reader Version: 4.9.35

## Additional Context

I didn't have this issue today morning. I suppose the extension got updated and now I can't use pages in sidebar.
According to DevTools, pages in the sidebar have this (at least) one element from Dark Reader extension injected:
`<style class="darkreader darkreader--fallback" media="screen">html, body, body :not(iframe):not(div[style^="position:absolute;top:0;left:-"]) { background-color: #181a1b !important; border-color: #776e62 !important; color: #e8e6e3 !important; } html, body { opacity: 1 !important; transition: none !important; }</style>`
Deleting this element manually in DevTools fixes the problem, until next page load, obviously.

I can also see this log in the Console of the Dark Reader extension, not sure if it's relevant to the issue:
```
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'id')
    at reply (index.js:2843)
    at index.js:2853
```

**Comment by bershanskiy** [comment]
@jenda69 I'm having trouble preproducing this bug.  I opened Wikipedia (one of default entries) and `https://example.com` in sidebar panel, but Dark Reader was not able to do anything at all. Are you sure extensions are permitted to interact with sidebar panels?

**Comment by mateuszkozakiewicz** [comment]
> @jenda69 I'm having trouble preproducing this bug. I opened Wikipedia (one of default entries) and `https://example.com` in sidebar panel, but Dark Reader was not able to do anything at all. Are you sure extensions are permitted to interact with sidebar panels?

Usually no, but it is possible, a similar issue was once reported on Vivaldi forums: https://forum.vivaldi.net/topic/50265/dark-reader-effects-web-panels-uncontrollably

I can confirm this problem exists with current Vivaldi version 4.2.2406.48 and Dark Reader 4.9.35, toggling the dark mode off in Dark Reader settings doesn't help, only disabling the extension entirely.

![image](https://user-images.githubusercontent.com/8735263/134246647-455bee99-2104-40cd-a306-570969670246.png)

**Comment by bershanskiy** [comment]
On Chrome Web Store user called Vaelek Corvo posted the following:

> 9/21 update broke Vivaldi side panels
> All Vivaldi side panels as of the 9/21 update are stick in dark mode where before they wouldn't go dark even if you wanted them to.

So Vivaldi normally did not permit this and now it does for multiple users.

**Comment by mateuszkozakiewicz** [comment]
Also please note that the extension is not working correct in side panels (it would be awesome if it did) - see here for example facebook page opened in web panel (left) and normal tab (right), both are parsed by Dark Reader but look very different
![image](https://user-images.githubusercontent.com/8735263/134248711-416fa69f-601e-4ec8-90ad-d4bc7f28dbcb.png)

**Comment by bershanskiy** [comment]
What non-default settings do you have? What other extensions do you have installed? Perhaps, combination of multiple factors makes Vivaldi behave like this.

Based on your earlier error message, I concocted a possible "kill switch". Apparently, when content script sends a message from a pane, it does not have `tab` property. We could simply check for existence of it and if it's missing send response to stop execution of content script. The problem is, we can't send a new one-time message (because there is no tab id to send this to) and I'm not sure how reliable `sendResponse` is in this case.

**Comment by mateuszkozakiewicz** [comment]
> What non-default settings do you have? What other extensions do you have installed? Perhaps, combination of multiple factors makes Vivaldi behave like this.

I tried disabling all extensions except Dark Reader but it has no effect, my settings are default with "not invert listed" and a few exceptions added (facebook.com and whatsapp.com being among them, yet they are affected in web panels, not in tabs).
Problems started happening today after upgrading to newest Vivaldi version but rolling back doesn't fix it.

**Comment by aftadizca** [comment]
![image](https://user-images.githubusercontent.com/16900725/134264006-41dab041-0764-4c5d-b15d-bd06aa33a664.png)
i have similar issue... Got blank page on WhatappsWeb... 
![image](https://user-images.githubusercontent.com/16900725/134264104-ce6ff69b-dd9c-4f3e-93a3-4ef0e3021c22.png)
when disable this.. fixes the problem.. but when reload back to blank

**Comment by Gusted** [comment]
> The problem is, we can't send a new one-time message (because there is no tab id to send this to) and I'm not sure how reliable `sendResponse` is in this case.

https://github.com/darkreader/darkreader/blob/master/src/background/tab-manager.ts#L86
This condition should kick in for the Vivaldi panels. And then send a `unsupportedSender` back to the the panel, whereby the panel will clean up DR.

**Comment by mateuszkozakiewicz** [comment]
> @mateuszkozakiewicz If you have time, please test [this new version](https://github.com/bershanskiy/darkreader/actions/runs/1260859038) (download `artifact.zip`, extract Chrome build, and install it). If you need help, I can give more detailed instructions.

Hello, with this version the problem is fixed, web panels are not affected in any way by the extension.

**Comment by mateuszkozakiewicz** [comment]
Hello, 
I see the version in Chrome extension store is now updated however it didn't fix the problem (at least for me). [The version you told me](https://github.com/bershanskiy/darkreader/actions/runs/1260859038) to test still works fine and fixes the problem. Is there a difference between them?

**Comment by bershanskiy** [comment]
Thanks for info.

> I see the version in Chrome extension store is now updated however it didn't fix the problem (at least for me).

Current released version, 4.9.36, has this bug, so this is expected.

> The version you told me to test still works fine and fixes the problem.

Thanks for confirming.

> Is there a difference between them?

PR #6858 fixes the problem. It was not included in 4.9.36 (because it landed after 4.9.36 was released), but the next version (4.9.37) will have the fix.
